### ISSUE TITLE
fromJournal method in Typed Event Adapter should return a list of events 

### ISSUE THREAD

**Issue by michaltomanski** [akka/akka#26909]
Currently, the `EventAdapter` in Akka Typed is:
```
abstract class EventAdapter[Event, PersistedEvent] {
  def toJournal(e: Event): PersistedEvent
  def fromJournal(p: PersistedEvent): Event
}
```

I suggest changing the `fromJournal` signature to return a sequence of events, which would give us something like
```
abstract class EventAdapter[Event, PersistedEvent] {
  def toJournal(e: Event): PersistedEvent
  def fromJournal(p: PersistedEvent): Seq[Event]
}
```
Alternatively, we can use a specialized structure for the return type
```
sealed abstract class EventSeq[E] {
  def events: immutable.Seq[E]
}
```

This modification has several advantages:
* compatibility with untyped version of EventAdapter (where an `EventSeq` is returned by `fromJournal`method)
* possibility to handle schema evolution case where one event has to be split into two, as described in [Akka docs](https://doc.akka.io/docs/akka/current/persistence-schema-evolution.html#split-large-event-into-fine-grained-events)
* possibility to ignore events (by returning an empty list), as described in [Akka docs](https://doc.akka.io/docs/akka/current/persistence-schema-evolution.html#remove-event-class)

This request is a follow up to the discussion:
https://discuss.lightbend.com/t/recommended-approach-to-unused-events-with-akka-typed-persistence/4159

**Comment by patriknw** [comment]
It was added in https://github.com/akka/akka/pull/25050 but I can't see that we discussed that matter. Maybe we were looking for simplest possible API, but went too far in that direction.

Any thoughts @chbatey ?

**Comment by chbatey** [comment]
I recall us discussing simplifying it but nothing specific, I have nothing against making it a Seq

**Comment by michaltomanski** [comment]
Thanks @patriknw !

**Comment by patriknw** [comment]
thanks for reporting, will be included in 2.6.0-M4 that is to be released today
